src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
-		union = polygonUnion(polys, buf=buf)
-		orientation = VectorField.forUnionOf(regs)
-		return PolygonalRegion(polygon=union, orientation=orientation)
-
-	@property
-	def boundary(self) -> PolylineRegion:
-		"""Get the boundary of this region as a `PolylineRegion`."""
-		return PolylineRegion(polyline=self.polygons.boundary)
-
-	def buffer(self, distance):
-		return PolygonalRegion(polygon=self.polygons.buffer(distance))
-
-	@cached_property
-	def prepared(self):
-		return shapely.prepared.prep(self.polygons)
-
-	def containsPoint(self, point):
-		return self.prepared.intersects(shapely.geometry.Point(point))
-
-	def containsObject(self, obj):
-		objPoly = obj.polygon
-		if objPoly is None:
-			raise ValueError('tried to test containment of symbolic Object!')
-		# TODO improve boundary handling?
-		return self.prepared.contains(objPoly)
-
-	def containsRegion(self, other, tolerance=0):
-		if isinstance(other, EmptyRegion):
-			return True
-		poly = toPolygon(other)
-		if poly is None:
-			raise TypeError(f'cannot test inclusion of {other} in PolygonalRegion')
-		return self.polygons.buffer(tolerance).contains(poly)
-
-	@distributionMethod
-	def distanceTo(self, point):
-		return self.polygons.distance(shapely.geometry.Point(point))
-
-	def getAABB(self):
-		xmin, ymin, xmax, ymax = self.polygons.bounds
-		return ((xmin, ymin), (xmax, ymax))
-
-	def show(self, plt, style='r-', **kwargs):
-		plotPolygon(self.polygons, plt, style=style, **kwargs)
-
-	def __repr__(self):
-		return f'PolygonalRegion({self.polygons!r})'
-
-	def __eq__(self, other):
-		if type(other) is not PolygonalRegion:
-			return NotImplemented
-		return (other.polygons == self.polygons
-		        and other.orientation == self.orientation)
-
-	@cached
-	def __hash__(self):
-		# TODO better way to hash mutable Shapely geometries? (also for PolylineRegion)
-		return hash((str(self.polygons), self.orientation))
-
-	def __getstate__(self):
-		state = self.__dict__.copy()
-		state.pop('_cached_prepared', None)		# prepared geometries are not picklable
-		return state
+    """Region given by one or more polygons (possibly with holes).
+
+    The region may be specified by giving either a sequence of points defining the
+    boundary of the polygon, or a collection of ``shapely`` polygons (a ``Polygon``
+    or ``MultiPolygon``).
+
+    Args:
+        points: sequence of points making up the boundary of the polygon (or `None` if
+            using the **polygon** argument instead).
+        polygon: ``shapely`` polygon or collection of polygons (or `None` if using
+            the **points** argument instead).
+        orientation (`VectorField`; optional): :term:`preferred orientation` to use.
+        name (str; optional): name for debugging.
+    """
+    def __init__(self, points=None, polygon=None, orientation=None, name=None, z=0, additionalDeps=[]):
+        super().__init__(name, *additionalDeps, orientation=orientation)
+
+        self.z = z
```

Comment:
Not documented.

1-line summary in the docstring should probably include "at a fixed Z coordinate".